File: src/main/java/pack/web/param/AccountRq.java
package pack.web.param;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class AccountRq implements Rq {

    private final String requestId;

    @JsonCreator
    public AccountRq(@JsonProperty(value = "request_id", required = true) String requestId) {
        this.requestId = requestId;
    }

    public String getRequestId() {
        return requestId;
    }

    @Override
    public List<String> getLocks() {
        return List.of(requestId);
    }
}

File: src/main/kotlin/pack/web/param/AccountRq.kt
package pack.web.param

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import pack.web.Validator.requireNotBlank

class AccountRq @JsonCreator constructor(@JsonProperty("request_id") requestId: String?) : Rq {
    val requestId: String = requireNotBlank(requestId, "request_id")

    override val locks: List<String>
        get() = listOf(requestId)
}
